### DISCUSSION TITLE
Please, help! Can't get it to work!

### DISCUSSION THREAD

**Discussion by xilopaint** [FabCx/TradingView-Webhook-Bot#46] [General]
I have followed all the steps of the installation section from 1 to 8. The project is hosted locally and after running `python3 main.py` nothing happens once my configured alert is fired on TradingView. I get no errors on Terminal and the python process just keeps running with no messages. I already tried to change the port on `main.py` to 443 and no luck.

Should I get the `Alert Received & Sent!` message after the alert is fired, I presume? What could be the problem?

**Comment by fabston** [comment]
Hi,

Do you run the bot on your own local machine or on a VPS?

And yes - `Alert Received & Sent!` will be shown after the alert triggered on TradingView.

**Reply by xilopaint** [reply]
On my local machine.

**Reply by xilopaint** [reply]
Please, feel free to make any obvious questions. I might be missing something obvious. I can send screenshots if necessary.

**Reply by fabston** [reply]
It is not recommend to run the bot on a local machine. Since there are so many different factors I am not providing support for it. I've mentioned a few things to look at here #39. Sorry about that.

**Reply by xilopaint** [reply]
Hey @fabston, I got closer so maybe now you can help me. First I've set my router like this:

[EDITED]

I tested the webhook using this:

`curl -H 'Content-Type: application/json' -d '{"key": "abc", "msg": "foo"}' <MY_EXTERNAL_IP>/webhook`

It worked from within my network and I got the `Sent alert` confirmation but it didn't work from outside my network.

My second attempt was setting the router with port 8080 on the field for the public TCP port and port 80 on the field for the private TCP port:

[EDITED]

I used `curl` again but at this time mentioning the port 8080:

`curl -H 'Content-Type: application/json' -d '{"key": "abc", "msg": "foo"}' <MY_EXTERNAL_IP>:8080/webhook`

The `curl` command worked both from inside and outside my network and I got the confirmation, but alerts from the TradingView website still were not received by the bot. This is how I've set the alert on TradingView:

[EDITED]

TradingView doesn't let me use http://<MY_EXTERNAL_IP>:8080/webhook as a URL because it only accepts port 80.

Any thoughts?

**Reply by fabston** [reply]
Hi,

That is right, TradingView only allows port `80` & `443`. You'd either have to run the bot behind port 80 (be sure to open that port up in your router) or you forward port 8080 to 80 using ufw, as an example.

**Reply by xilopaint** [reply]
> Hi,
> 
> That is right, TradingView only allows port `80` & `443`. You'd either have to run the bot behind port 80 (be sure to open that port up in your router) or you forward port 8080 to 80 using ufw, as an example.

[EDITED]

Isn't the above equal to forward port 80 to 8080? Or are you suggesting me to do the opposite? I'm lost.

On README you say: "It is recommended to run flask on a different port like 8080. It is then necessary to forward port 80 to 8080".

**Reply by fabston** [reply]
Forgive me, mixed it up. Yes - you'd either have to forward port 80 to 8080 with something like ufw. 

The image you shared, is that from your router settings? 

As mentioned before, running it locally is always tricky and that's why it's not recommend hence no support.

**Reply by xilopaint** [reply]
> The image you shared, is that from your router settings?

Yes.

I feel I'm so close!!!

**Reply by xilopaint** [reply]
> Forgive me, mixed it up. Yes - you'd either have to forward port 80 to 8080 with something like ufw.

Do you mean forwarding the port on my router settings is not correct and I should use something like ufw? Ou should I use both?

**Reply by fabston** [reply]
The port forwarding must be done on your machine, not router. On Linux [UFW](https://help.skysilk.com/support/solutions/articles/9000182486--basic-forward-a-port-with-ufw-in-ubuntu-18-04-linux) would be an option. On your router only poart 80 must be opened.

**Reply by xilopaint** [reply]
[EDITED]

Do you have any idea why using the above on my router makes the `curl` command work from inside my network but not from outside?

The `curl` command in this case is the following:

`curl -H 'Content-Type: application/json' -d '{"key": "abc", "msg": "foo"}' <MY_EXTERNAL_IP>/webhook`
